src/hardware/Relay.h

```diff
@@ -0,0 +1,26 @@
+/**
+ * @file Relay.h
+ * 
+ * @brief A relay connected to a GPIO pin
+ */
+#pragma once
+
+class GPIOPin;
+
+enum RelayTriggerType {
```

Comment:
Really just a matter of taste but I tend to define these inside the class:

```cpp
class Relay {
public:
  enum TriggerType {
    LOW_TRIGGER,
    HIGH_TRIGGER,
  };

  Relay(/*...*/);
  /* ... */
private:
  /* ... */
}
```

That way you don't have the doubling of `Relay` in the name. Inside the class you refer to it as `TriggerType` and if you need it outside, it's `Relay::TriggerType`.